go/interpreter/lfvm/instructions.go

```diff
@@ -330,27 +299,8 @@ func opCallDatasize(c *context) {
 
 func opCallDataload(c *context) {
 	top := c.stack.peek()
-	if !top.IsUint64() {
-		top.Clear()
-		return
-	}
-
-	offset := top.Uint64()
-	input := c.params.Input
-	var value [32]byte
-	for i := 0; i < 32; i++ {
-		pos := i + int(offset)
-		if pos < 0 {
-			top.Clear()
-			return
-		}
-		if pos < len(input) {
-			value[i] = input[pos]
-		} else {
-			value[i] = 0
-		}
-	}
-	top.SetBytes(value[:])
+	value := getData(c.params.Input, top, 32)
```

Comment:
this could lead to an out of memory panic, this is why `getData` was guarded by `getSlice` before.

Reply:
This call cannot lead to any panic. Other than the allocator failing to allocate 32 bytes. 
the explanation is in the pr description.